Arrowgene.Ddon.GameServer/Handler/CraftStartEquipGradeUpHandler.cs

```diff
@@ -0,0 +1,273 @@
+            {
+                Unk0 = 0,          // Probably DragonAugment related.
+                Unk1 = 0,
+                Unk2 = 0,          // setting this to a value above 0 seems to stop displaying "UP" ?
+                Unk3 = 0,          // displays "UP" next to the DA upon succesful enhance.
+                IsMax = false,      // displays Max on the DA popup.
+            };
+            CDataCraftStartEquipGradeUpUnk0 dummydata = new CDataCraftStartEquipGradeUpUnk0()
+            {
+                Unk0 = new List<CDataCraftStartEquipGradeUpUnk0Unk0> { DragonAugmentData },
+                Unk1 = 0,
+                Unk2 = 0,
+                Unk3 = 0,               // No idea what these 3 bytes are for
+                DragonAugment = false,    // makes the DragonAugment slot popup appear if set to true.
+            };
+            // TODO: Source these values accurately when we know what they are. ^
+
+            List<CDataItemUpdateResult> updateResults;
+            var res = new S2CCraftStartEquipGradeUpRes();
+            S2CItemUpdateCharacterItemNtc updateCharacterItemNtc = new S2CItemUpdateCharacterItemNtc();
+
+            #endregion
+
+            #region Point Distribution 
+            // Handles adding EquipPoints.
+            // These values for adding equippoints are completely made up and not representative of any real Pawn craft levels.
+            if(IsGreatSuccess == true)
+            {
+                addEquipPoint = 300;
+                addEquipPoint = (uint)(addEquipPoint * pointsMultiplier);
+                currentTotalEquipPoint += addEquipPoint;
+                bool updateSuccessful = Server.Database.UpdateItemEquipPoints(equipItemUID, currentTotalEquipPoint);
+            }
+            else
+            {
+                addEquipPoint = 180;
+                addEquipPoint = (uint)(addEquipPoint * pointsMultiplier);
+                currentTotalEquipPoint += addEquipPoint;
+                bool updateSuccessful = Server.Database.UpdateItemEquipPoints(equipItemUID, currentTotalEquipPoint);
+            }
+
+            #endregion
+
+            // TODO: we need to implement Pawn craft levels since that affects the points that get added
+
+
+            #region Taking Cost
+            // Removes crafting materials
+            foreach (var craftMaterial in request.CraftMaterialList)
+            {
+                updateResults = _itemManager.ConsumeItemByUIdFromMultipleStorages(Server, client.Character, ItemManager.BothStorageTypes, craftMaterial.ItemUId, craftMaterial.ItemNum);
+                updateCharacterItemNtc.UpdateItemList.AddRange(updateResults);
+            }
+
+            // Subtract less Gold if supportpawn is used and add slightly more points
+            if(request.CraftSupportPawnIDList.Count > 0)
+            {
+                goldRequired = (uint)(goldRequired*0.95);
```

Comment:
Where did this logic come from?

Reply:
Observating the client, it's what it does with the price. It's the same in the regular crafting handler